### PR TITLE
Added groups to catalogs

### PR DESCRIPTION
**Pull request by AlmightyMikkel**

### CODE REVIEW

File: Runtime/Game/Requests/CatalogRequests.cs [hunk 1]
```diff
@@ -349,6 +403,8 @@ public class LootLockerListCatalogPricesResponse : LootLockerResponse
         /// </summary>
         public Dictionary<string /*catalog_listing_id*/, LootLockerCurrencyDetails> currency_details { get; set; }
 
+        public Dictionary<string /*catalog_listing_id*/, LootLockerGroupDetails> group_details { get; set; }
```
Comment (kirre-bylund) [inline Runtime/Game/Requests/CatalogRequests.cs:406]: You've missed adding group_details to the AppendCatalogItems method

File: Runtime/Game/Requests/CatalogRequests.cs [hunk 2]
```diff
@@ -511,7 +582,8 @@ public LootLockerInlinedCatalogEntry[] GetLootLockerInlinedCatalogEntries()
                     LootLockerCatalogEntryEntityKind.asset == entityKind && asset_details.ContainsKey(catalogListingID) ? asset_details[catalogListingID] : null, 
                     LootLockerCatalogEntryEntityKind.progression_points == entityKind && progression_points_details.ContainsKey(catalogListingID) ? progression_points_details[catalogListingID] : null, 
                     LootLockerCatalogEntryEntityKind.progression_reset == entityKind && progression_resets_details.ContainsKey(catalogListingID) ? progression_resets_details[catalogListingID] : null, 
-                    LootLockerCatalogEntryEntityKind.currency == entityKind && currency_details.ContainsKey(catalogListingID) ? currency_details[catalogListingID] : null
+                    LootLockerCatalogEntryEntityKind.currency == entityKind && currency_details.ContainsKey(catalogListingID) ? currency_details[catalogListingID] : null,
+                    LootLockerCatalogEntryEntityKind.group == entityKind && group_details.ContainsKey(catalogListingID) ? group_details[catalogListingID] : null
```
Comment (kirre-bylund) [inline Runtime/Game/Requests/CatalogRequests.cs:586]: We need to chat about this. I have a vague recollection about this not going to work. It was something about us now needing a combined key of catalog_listing_id and something else. So let's chat about it.

**Review by kirre-bylund — CHANGES_REQUESTED** [review]

File: Runtime/Game/Requests/CatalogRequests.cs (hunk 1 above)
Comment (AlmightyMikkel) [reply Runtime/Game/Requests/CatalogRequests.cs:406]: Ty I added it now!

**Review by kirre-bylund — APPROVED** [review]
Approved. We need to remember though that this is a breaking change that I can't really figure out how we do a deprecation cycle on. Let's talk about communication around it.

File: Runtime/Game/Requests/CatalogRequests.cs [hunk 3]
```diff
@@ -416,70 +585,90 @@ public LootLockerListCatalogPricesResponse(LootLockerResponse serverResponse)
...
+                foreach (var detail in parsedResponse.assets_details)
                 {
-                    asset_details[assetDetail.catalog_listing_id] = assetDetail;
+                    UnityEngine.Debug.Log(detail.GetItemDetailsKey().catalog_listing_id);
+                    UnityEngine.Debug.Log(detail.GetItemDetailsKey().item_id);
+
+                    UnityEngine.Debug.Log(detail.catalog_listing_id);
+                    UnityEngine.Debug.Log(detail.id);
+
+
```
Comment (kirre-bylund) [inline Runtime/Game/Requests/CatalogRequests.cs:597]: Remove

**Review by kirre-bylund — APPROVED** [review]
Minor remark, otherwise this looks good.

**Pull request merged**